packages/uma-auth-client/src/hooks/useNwcRequester.tsx

```diff
@@ -1,16 +1,38 @@
 import { useEffect, useState } from "react";
+import * as Nip47 from "src/Nip47Types";
 import { NwcRequester } from "src/NwcRequester";
 import { useOAuth } from "./useOAuth";
 
+class TokenRefreshNwcRequester extends NwcRequester {
+  constructor(
+    nwcConnectionUri: string,
+    private tokenRefresh: () => Promise<void>,
```

Comment:
Why do we need a subclass for this instead of just injecting this into the requester? I try to avoid inheritance unless it's really the only way to do what we want.

Reply:
Updated to just inject into the NwcRequester and make the refreshToken request in _checkConnected

Will need to make a refreshToken function for non-react apps later (since this just uses the oAuthTokenExchange from useOAuth)